packages/babel-plugin-transform-runtime/scripts/build-dist.js

```diff
@@ -108,51 +108,74 @@ function writeCorejsExports(pkgDirname, runtimeRoot, paths) {
   outputFile(pkgJsonPath, JSON.stringify(pkgJson, undefined, 2) + "\n");
 }
 
-function writeHelpers(runtimeName, { corejs } = {}) {
-  const helperPaths = writeHelperFiles(runtimeName, { corejs, esm: false });
-  const helperESMPaths = writeHelperFiles(runtimeName, { corejs, esm: true });
-  writeHelperExports(runtimeName, helperPaths.concat(helperESMPaths));
+function writeHelperFile(
+  runtimeName,
+  pkgDirname,
+  helperPath,
+  helperName,
+  { esm, corejs }
+) {
+  const filePath = path.join(helperPath, esm ? "index.mjs" : "index.js");
+  const fullPath = path.join(pkgDirname, filePath);
+
+  outputFile(
+    fullPath,
+    buildHelper(runtimeName, pkgDirname, fullPath, helperName, { esm, corejs })
+  );
+
+  return `./${filePath}`;
 }
 
-function writeHelperExports(runtimeName, helperPaths) {
+function writeHelperLegacyESMFile(pkgDirname, helperName) {
+  const fullPath = path.join(pkgDirname, "helpers", "esm", `${helperName}.js`);
+
+  outputFile(fullPath, `export { default } from "../${helperName}/index.mjs"`);
+}
+
+function writeHelpers(runtimeName, { corejs } = {}) {
+  const pkgDirname = getRuntimeRoot(runtimeName);
   const helperSubExports = {};
-  for (const helperPath of helperPaths) {
-    helperSubExports[helperPath.replace(".js", "")] = helperPath;
+  for (const helperName of helpers.list) {
+    const helperPath = path.join("helpers", helperName);
+    helperSubExports[`./${helperPath}`] = {
+      module: writeHelperFile(runtimeName, pkgDirname, helperPath, helperName, {
+        esm: true,
+        corejs,
+      }),
+      node: writeHelperFile(runtimeName, pkgDirname, helperPath, helperName, {
+        esm: false,
+        corejs,
+      }),
+      get default() {
+        return this.module;
+      },
+    };
+    writeHelperLegacyESMFile(pkgDirname, helperName);
   }
+
+  writeHelperExports(runtimeName, helperSubExports);
+}
+
+function writeHelperExports(runtimeName, helperSubExports) {
   const exports = {
-    "./helpers/": "./helpers/",
     ...helperSubExports,
     "./package": "./package.json",
     "./package.json": "./package.json",
     "./regenerator": "./regenerator/index.js",
+    "./regenerator/*.js": "./regenerator/*.js",
+    "./helpers/esm/*": "./helpers/esm/*.js",
+    // These patterns are deprecated, but since patterns
+    // containing * are not supported in every Node.js
+    // version we keep them for better compatibility.
```

Comment:
I can't follow. Why should we add `*.js` if they are _not_ supported in every Node.js version? Is that for compatibility with other bundlers?

Reply:
It's that because the version without `*.js` is deprecated and it will not be supported by future Node.js versions. (`*.js` is the "standard" way of doing it)

Reply:
https://nodejs.org/api/packages.html#packages_subpath_folder_mappings